src/context/AuthProvider.js

```diff
@@ -0,0 +1,53 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { appReducer } from "../reducers";
+import {
+  createUserWithEmailAndPassword,
+  onAuthStateChanged,
+  signInWithEmailAndPassword,
+  signOut,
+} from "firebase/auth";
+import { auth } from "../firebase.config";
+
+const AuthContext = createContext();
+
+export const AuthProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  const signUpHandler = async (email, password) => {
+    return createUserWithEmailAndPassword(auth, email, password);
+  };
+
+  const loginHandler = async (email, password) => {
+    return signInWithEmailAndPassword(auth, email, password);
+  };
+
+  const logout = () => {
+    return signOut(auth);
+  };
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
```

Comment:
setLoading(true) might be required here, or is it already set elsewhere?

Reply:
na na, loading will be false here. This is required because firebase-auth will set the token, localstorage, etc. So we need to worry about the initial check here, where the firebase checks if a user is already logged in or not. In either case, we will set the loading false.

Reply:
😅 I meant at starting. 
"setLoading(true)"---->setContext----->setLoading(false)..
Anyways I saw it is there at the top of the component.